lang/java/avro/src/main/java/org/apache/avro/file/DataFileReader12.java

```diff
@@ -61,6 +61,7 @@ public DataFileReader12(SeekableInput sin, DatumReader<D> reader) throws IOExcep
     this.in = new DataFileReader.SeekableInputStream(sin);
 
     byte[] magic = new byte[4];
+    in.seek(0); // seek to 0 to read magic header
     in.read(magic);
```

Comment:
what if fewer bytes come back? why not a readFully()?

Reply:
SeekableInputStream extends InputStream and doesn't have readFully.

Reply:
If fewer bytes come back, `read()` returns another number than 4 (this is discarded), and the leftover part of the array contains zeroes.
This fails the magic check later.

Reply:
since magic has been already initialized to 4 bytes, partial reads will anyways not match the magic. This wouldn't be an issue.